client/state/data-layer/wpcom-http/actions.js

```diff
@@ -30,15 +40,49 @@ export const http = ( {
 	onFailure,
 	onProgress,
 	...options,
-}, action = null ) => ( {
-	type: WPCOM_HTTP_REQUEST,
-	body,
-	method,
-	path,
-	query: { ...query, apiVersion },
-	formData,
-	onSuccess: onSuccess || action,
-	onFailure: onFailure || action,
-	onProgress: onProgress || action,
-	options,
-} );
+}, action = null ) => {
+	const actionProperties = {
+		onSuccess: onSuccess || action,
+		onFailure: onFailure || action,
+		onProgress: onProgress || action,
+		query: { ...query }
+	};
+
+	// * check parameters *
+
+	// `path` is not optional and it is a string
+	// `method` is not optional and its value can take either 'GET' or 'POST`
+	if (
+		( ! path || typeof path !== 'string' ) ||
+		( ! method || ( method !== 'GET' && method !== 'POST' ) )
+	) {
+		return badRequest();
+	}
+
+	// - One of `apiVersion` of `apiNamespace` should be defined
+	// - `apiVersion` and `apiNamespace` cannot be defined simultaneously
+	if (
+		( typeof apiVersion === 'undefined' && typeof apiNamespace === 'undefined' ) ||
+		( typeof apiVersion !== 'undefined' && typeof apiNamespace !== 'undefined' )
+	) {
+		return badRequest();
+	}
+
+	// - `apiVersion` must have the `v1, v2, ...` shape
+	if ( typeof apiVersion !== 'undefined' && ( /^v\d+/ ).test( apiVersion ) ) {
+		actionProperties.query.apiVersion = apiVersion;
+	}
+
+	// - `apiNamesapce` must have the `wp/v2, wpcom/v2, ...` shape
```

Comment:
>  let's get it clear in some documentation and make it obvious for people wanting to use this code.

Exactly this 👍 Though I'm sure you had some PRs planned for this @retrofox and @nylen ?

Perhaps we should put this PR on hold until we get some better documentation PRs in wpcom.js and have it be much more obvious and strict when we give it bad arguments? I'd be happy to take a look and help test.

Does something like this order make sense?
- wpcom.js documentation PR
- wpcom.js validation + version bump PR
- bump wpcom.js in Calypso, with full regression testing.
- update wpcom-http to accept apiNamespace PR
- (optional) eslint rule / calypso integration